api/src/main/java/org/openmrs/module/registrationcore/api/mpi/common/MpiException.java

```diff
@@ -0,0 +1,28 @@
+package org.openmrs.module.registrationcore.api.mpi.common;
+
+/**
+ * Represents errors that occur during interactions with the Master Patient Index.
+ */
+public class MpiException extends RuntimeException {
```

Comment:
By convention, this being a runtime exception, it should extend APIException instead.

Reply:
Changed, would you mind explaining runtime exception vs API exception and when we should use / extend which? (Just for my understanding)

Reply:
Because APIException is already a subclass of RuntimeException so you should extend it instead because your code benefits from areas that generically handle APIExceptions which won't happen for any other subclass of RuntimeException

Reply:
ok thank you!